src/Idle.Logic/Languages/LanguagesViewModel.cs

```diff
@@ -28,5 +29,17 @@ public async Task LoadAsync()
 			Languages.AddRange(languageViewModels);
 
 		}
+
+		public async Task PushAsync()
+		{
+			var languages = new List<Language>();
+			foreach(LanguageViewModel vm in Languages)
+            {
+				languages.Add(vm._language);
+            }
+
+			await _languageRepository.UpdateAllAsync(languages);
```

Comment:
looks ok.
what do you think of this?
you should at least return the task without awaiting it. 
await is not required here

```cs
public Task<int> PushAsync()
{
	var languages = Languages.Select(vm => vm._language);
	return _languageRepository.UpdateAllAsync(languages);
}
```

Reply:
Yes this looks a lot better, thanks brayan